src/main/java/thePackmaster/patches/MetricsPatches.java

```diff
@@ -0,0 +1,110 @@
+            if (AbstractDungeon.player.chosenClass == THE_PACKMASTER) {
+                httpRequest.setMethod("PUT");
+            }
+        }
+
+        private static class Locator extends SpireInsertLocator {
+            @Override
+            public int[] Locate(CtBehavior method) throws Exception {
+                Matcher matcher = new Matcher.MethodCallMatcher(Net.HttpRequest.class, "setContent");
+                return LineFinder.findInOrder(method, matcher);
+            }
+        }
+    }
+
+
+    @SpirePatch(clz = GameOverScreen.class, method = "shouldUploadMetricData")
+    public static class ShouldUploadMetricData {
+        @SpirePostfixPatch
+        public static boolean patch(boolean returnValue) {
+            if (AbstractDungeon.player.chosenClass == THE_PACKMASTER) {
+                returnValue = Settings.UPLOAD_DATA;
+            }
+            return returnValue;
+        }
+
+    }
+
+    @SpirePatch(clz = Metrics.class, method = "run")
+    public static class RunPatch {
+        @SpirePostfixPatch
+        public static void patch(Metrics metrics) {
+            if (metrics.type == Metrics.MetricRequestType.UPLOAD_METRICS && AbstractDungeon.player.chosenClass == THE_PACKMASTER) {
+                try {
+                    Method addDataMethod = Metrics.class.getDeclaredMethod("addData", Object.class, Object.class);
+                    addDataMethod.setAccessible(true);
+                    addDataMethod.invoke(metrics, "pmversion", findTheModVersion());
+                    addDataMethod.invoke(metrics, "currentPacks", SpireAnniversary5Mod.currentPoolPacks.stream().map(p -> p.packID).collect(Collectors.joining(",")));
+
+                    String filteredPacks = SpireAnniversary5Mod.unfilteredAllPacks.stream()
+                            .filter(p -> !SpireAnniversary5Mod.allPacks.contains(p))
+                            .map(p -> p.packID)
+                            .collect(Collectors.joining(","));
+                    addDataMethod.invoke(metrics, "filteredPacks", filteredPacks);
+
+                    //Pack choices at the beginning formatted as Pack1|*Pack2|Pack3,Pack1... * signifying the selected pack
+                    addDataMethod.invoke(metrics, "packChoices", String.join(",", packChoices));
+                    boolean draftEnabled = SpireAnniversary5Mod.getCustomDraftEnabled();
+                    addDataMethod.invoke(metrics, "customDraftEnabled", draftEnabled);
+                    if(draftEnabled)
+                        addDataMethod.invoke(metrics, "customDraftConfiguration", SpireAnniversary5Mod.modConfig.getString("PackmasterCustomDraftSelection"));
+
+                    addDataMethod.invoke(metrics, "pickedHat", SpireAnniversary5Mod.getLastPickedHatID());
+                    addDataMethod.invoke(metrics, "enabledExpansionPacks", Loader.isModLoaded("expansionPacks")); //TODO: Actually use the correct field once both branches are merged
```

Comment:
Making the ID of the expansion mod a constant in the main mod file would be good. Maybe you've already done that on the branch where you're working on it, but still good to do it here too! Easier to change all references if the name changes during development that way.